### PR TITLE
Fixes EqualWeightingPortfolioConstructionModel not flattening expired insights

### LINKED ISSUE #2380: Equal Weighting Portfolio Model Not Flattening Expired Insights
#### Expected Behavior
When an insight expires (time > start+period) it the portfolio construction model should remove the position.

#### Actual Behavior
EWPCM holds the position until another one comes along.

#### Potential Solution
Check expirations flatten positions independent of other insights. 

#### Reproducing the Problem
And short timeframe alpha model will appear to hold like a portfolio.
 
#### Checklist 
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by AlexCatarino**
#### Description
- When there are or aren't new insights, the EqualWeightingPortfolioConstructionModel will creates a target to flatten delisted securities from the universe of expired insights.
- Helper methods were added to deal with removing expired insights and getting active ones and used in `EqualWeightingPortfolioConstructionModel`
- Adds unit test
- Updates framework algorithms

#### Related Issue
Closes #2380 

#### Motivation and Context
Fixes bugs in EqualWeightingPortfolioConstructionModel.

#### Requires Documentation Change
Yes. We should state in documentation that this model will remove expired insights.

#### How Has This Been Tested?
Unit and regression tests. 

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs [hunk 1]
```diff
@@ -40,50 +53,55 @@ public class EqualWeightingPortfolioConstructionModel : PortfolioConstructionMod
...
-            targets.AddRange(from kvp in algorithm.Portfolio
-                             where kvp.Value.Invested
-                             where targets.All(x => kvp.Key != x.Symbol)
-                             select new PortfolioTarget(kvp.Key, 0));
+            // Get expired insights and create flatten targets for each symbol
+            var expiredInsights = _insightCollection.RemoveExpiredInsights(algorithm.UtcTime);
+
+            var expiredTargets = from insight in expiredInsights
+                                 group insight.Symbol by insight.Symbol into g
+                                 where !_insightCollection.HasActiveInsights(g.Key, algorithm.UtcTime)
```
Comment (Martin-Molinero) [inline Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs]: Is this check require? Looks like we untrust the return value of `RemoveExpiredInsights`
Also, I was thinking that maybe moving the call to `RemoveExpiredInsights` before `GetActiveInsights` would avoid some double work

File: Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs [hunk 2]
```diff
@@ -40,50 +53,55 @@ public class EqualWeightingPortfolioConstructionModel : PortfolioConstructionMod
...
-            if (_removedSymbols != null)
-            {
-                // zero out securities removes from the universe
-                targets.AddRange(_removedSymbols.Select(s => new PortfolioTarget(s, 0)));
-                _removedSymbols = null;
-            }
 
-            if (insights.Length == 0)
+            if (algorithm.UtcTime < _nextExpiryTime &&
+                algorithm.UtcTime < _rebalancingTime &&
```
[the remark this answers is not in the document]
Comment (AlexCatarino) [reply Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs]: We added a `DateTime.Tick` to the returned result of `InsightCollection.GetNextExpiryTime` to keep that guard cause.

**Comment by mchandschuh** [comment]
Please consider the impact of this PR on these changes: https://github.com/QuantConnect/Lean/pull/2385

**Comment by jaredbroad** [comment]
One of the more complex refactors to date despite being a fairly simple model. The EWPCM considers a number of points in this refactor:

 - When an asset is delisted from the universe; we should remove it from the insights collection and send a signal to the execution model to trigger a position flattening.

- When there are multiple signals for different periods we should allocate accordingly to the one which is expiring soonest. E.g. 1-day downtrend combined with a 1-minute uptrend. This allows people to manually trigger short-term insights for scalping style models while also being compatible with long-term rebalancing strategies.

 - Even when there are no insight changes; the weighting of the assets can change significantly over a few days as the prices move. We needed a rebalancing _period_ in addition to the simple rebalancing on insight expirations.

 - When insights expire; only flatten the position when there are no _other active_ insights in the collection. Be smarter about handing those expirations to minimize fees.

Thanks to @AlexCatarino @mchandschuh

**Pull request merged**